service/src/main/java/io/camunda/service/license/SaaSLicense.java

```diff
@@ -7,8 +7,12 @@
  */
 package io.camunda.service.license;
 
-public class EnvironmentVariableReader {
-  public String getEnvironmentVariableValue(final String envVarName) {
-    return System.getenv().getOrDefault(envVarName, "");
+public class SaaSLicense extends CamundaLicense {
```

Comment:
`SaaSLicense` has nothing in common with `CamundaLicense` expect overriding `isValid()` method
I think it is better to extract an interface `CamundaLicense` with `isValid()` method and create a `SelfManagedCamundaLicense` that implements it with all the license checks, and create a second implementation of `CamundaLicense`: `SaaSLicense` or `NoopLicense` that just implements  `isValid()`.
This way you avoid `SaaSLicense` accidentelly having access to public or protected methods of `CamundaLicense`.